x-pack/plugins/lens/public/datatable_visualization/visualization.tsx

```diff
@@ -320,26 +339,56 @@ export const datatableVisualization: Visualization<DatatableVisualizationState>
           arguments: {
             title: [title || ''],
             description: [description || ''],
-            columns: columns.map((column) => ({
-              type: 'expression',
-              chain: [
-                {
-                  type: 'function',
-                  function: 'lens_datatable_column',
-                  arguments: {
-                    columnId: [column.columnId],
-                    hidden: typeof column.hidden === 'undefined' ? [] : [column.hidden],
-                    width: typeof column.width === 'undefined' ? [] : [column.width],
-                    isTransposed:
-                      typeof column.isTransposed === 'undefined' ? [] : [column.isTransposed],
-                    transposable: [
-                      !datasource!.getOperationForColumnId(column.columnId)?.isBucketed,
-                    ],
-                    alignment: typeof column.alignment === 'undefined' ? [] : [column.alignment],
+            columns: columns.map((column) => {
+              const paletteConfig = column.palette?.params;
+
+              return {
+                type: 'expression',
+                chain: [
+                  {
+                    type: 'function',
+                    function: 'lens_datatable_column',
+                    arguments: {
+                      columnId: [column.columnId],
+                      hidden: typeof column.hidden === 'undefined' ? [] : [column.hidden],
+                      width: typeof column.width === 'undefined' ? [] : [column.width],
+                      isTransposed:
+                        typeof column.isTransposed === 'undefined' ? [] : [column.isTransposed],
+                      transposable: [
+                        !datasource!.getOperationForColumnId(column.columnId)?.isBucketed,
+                      ],
+                      alignment: typeof column.alignment === 'undefined' ? [] : [column.alignment],
+                      colorMode: [column.colorMode ?? 'none'],
+                      palette: [
+                        {
+                          type: 'expression',
+                          chain: [
+                            {
+                              type: 'function',
+                              function: 'palette',
+                              arguments: {
```

Comment:
This logic should be part of the chart plugin - see how it's done in xy charts: https://github.com/elastic/kibana/blob/6b8c44d113efd7ca05efb0be7b71a2bda090f3a7/x-pack/plugins/lens/public/xy_visualization/to_expression.ts#L268